containers/unit-tests/start

```diff
@@ -1,33 +1,199 @@
-#!/bin/sh
-set -eu
-
-top_srcdir="$(realpath "${0%/*}/../..")"
-image=ghcr.io/cockpit-project/unit-tests:latest
-flags=''
-
-while test $# -gt 0; do
-  case "$1" in
-    --docker) docker="docker";;
-    --podman) docker="podman";;
-    CC=*) ccarg="--env=$1";;
-    shell) flags=-it; cmd=/bin/bash;;
-    :*) image=ghcr.io/cockpit-project/unit-tests"$1";;
-    build|dist|distcheck|check-memory) cmd="/source/containers/unit-tests/run.sh $1";;
-    *) echo "Unknown option '$1'"; exit 1;;
-  esac
-  shift
-done
-
-if test -z "${cmd:-}"; then
-  echo 'Must specify a scenario or `shell`'
-  exit 1
-fi
-
-if test -z "${docker:=$(which podman || which docker || true)}"; then
-  echo 'Neither podman nor docker are installed'
-  exit 1
-fi
-
-set -ex
-exec ${docker} run --shm-size=512M --volume "${top_srcdir}":/source:ro \
-       ${ccarg:+"$ccarg"} $flags -- "$image" $cmd
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import shlex
+import sys
+import tempfile
+
+from subprocess import run
+
+
+def logged(func):
+    def wrapper(args, **kwargs):
+        print('+', shlex.join(args))
+        return func(args, **kwargs)
+    return wrapper
+
+
+def git(*args):
+    run(['git', *args], check=True)
```

Comment:
I was looking for a `def run` here for quite a while until I realized that this is `subprocess.run`. Do you mind if we just use this qualified? I feel it's a too generic name for the top-level namespace. (*Not* a strong opinion)

Reply:
The issue is the clever hacking we do when `-v` is presented:

```python
def logged(func):
    def wrapper(args, **kwargs):
        print('+', shlex.join(args))
        return func(args, **kwargs)
    return wrapper

...later...

    if args.verbose:
        global run
        run = logged(run)
```